File: lab04/Quadrilateral.cpp
// File: Quadrilateral.h
// Author: Amandeep Gill
// Contents: the implementation of the Quadrilateral class

#include <string>
#include "Quadrilateral.h"

Quadrilateral::Quadrilateral()
{
    debug << "Entering default constructor for Quadrilateral\n";
    side = 0;
    debug << "Exiting default constructor for Quadrilateral\n";
}

string Quadrilateral::Sides() const
{
    debug << "Entering Sides function for Quadrilateral\n";
    debug << "Exiting Sides function for Quadrilateral\n";
    return "4";
}

File: lab08/Quadrilateral.java
// File: Quadrilateral.java
// Author: Amandeep Gill
// Contents: declaration and implementation of the virtual class Quadrilateral

public class Quadrilateral extends Shape {
    protected int side2;

    public Quadrilateral() {
    }

    public String sides() {
        return "4";
    }

    @Override
    public String getName() {
        return "Quadrilateral";
    }

    public void setSide2(int s2) {
        side2 = s2;
    }

    public int getSide2() {
        return side2;
    }
}
